packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -316,6 +317,38 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
       const abortController = new AbortController();
 
       if (isOnline) {
+        if (visitToEdit?.uuid) {
+          const existingAttributes = visitToEdit.attributes.map((attr) => attr.attributeType.uuid) || [];
+          const updatedAttributes = [];
+          const addedAttributes = [];
+          payload.attributes.forEach((attribute) => {
+            if (existingAttributes.includes(attribute.attributeType)) {
+              const attributeUuid = visitToEdit.attributes.filter(
+                (attr) => attr.attributeType.uuid === attribute.attributeType,
+              )[0].uuid;
+              updatedAttributes.push({ uuid: attributeUuid, value: attribute.value });
+            } else addedAttributes.push(attribute);
+          });
+          payload.attributes = addedAttributes;
+          for (const attribute of updatedAttributes) {
+            await openmrsFetch(`/ws/rest/v1/visit/${visitToEdit.uuid}/attribute/${attribute.uuid}`, {
+              method: 'POST',
+              headers: {
+                'Content-type': 'application/json',
+              },
+              body: {
+                value: attribute.value,
+              },
+            }).catch((err) => {
+              showSnackbar({
+                title: t('errorUpdatingVisitAttribute', 'Error updating visit attribute'),
+                kind: 'error',
+                isLowContrast: false,
+                subtitle: err?.message,
+              });
+            });
```

Comment:
Why are we updating the visit details even before we are updating the visit details. 

The right solution will be to first update visit details and then if the request is successful, then we update the attributes.